### DISCUSSION TITLE
WebView requests "Use camera"

### DISCUSSION THREAD

**Discussion by ltguillaume** [BrightDV/BoxBox#125] [Q&A]
For example:
1. Open _Race Hub_
1. Click the race, which opens planetf1.com's _On-Track Race Watchalong_ in a WebView

This requests "Use Camera" (as XPrivacyLUA shows me).

I believe WebViews can be further restricted from behavior like this.

**Comment by BrightDV** [comment]
Thanks for the report! I will have to investigate a bit.
However, the app itself can't have the permission to the Camera, as you can see in the [AndroidManifest.xml file](https://github.com/BrightDV/BoxBox/blob/07e88538b19c9957442dba649179ad12d50a6cc6/android/app/src/main/AndroidManifest.xml#L8) (internet access is the only permission of the app).
So, is it really a concern?

**Reply by ltguillaume** [reply]
Yeah, I'm not sure how this works. In theory, shouldn't be able to, but I was surprised it could still cause this warning, which is triggered if any of the methods `getCameraInfo`, `getNumberOfCameras`, `open`, `openLegacy` from the class `android.hardware.Camera` are called, or `openCamera` from `android.hardware.camera2.CameraManager` (source: https://github.com/M66B/XPrivacyLua/blob/master/app/src/main/assets/hooks.json)

**Reply by BrightDV** [reply]
Sorry for the late reply, I was really busy...
[I refactored a bit the webview](https://github.com/BrightDV/BoxBox/commit/b61861737e5b24091bbd8a805d0875ecb4aba6bc), so now it blocks all script/content requests that go not to the same domain, and blocks all the permission requests. 

Can you try to see it happens again? You can download the .apk from the latest Github action: https://github.com/BrightDV/BoxBox/actions/runs/7150510342.

(Enable experimental features in the settings and then open the article test screen from the drawer in order to open the webview).

Thanks!

**Reply by ltguillaume** [reply]
Thanks for looking into this! I think I only found this out during a race weekend with the _On-Track Race Watchalong_, so is there another way to open a similar site right now?

Either way, I think that _if_ https://github.com/BrightDV/BoxBox/commit/b61861737e5b24091bbd8a805d0875ecb4aba6bc doesn't interfere with any sources from CDNs, it would surely be a nicer way to restrict the website.

**Reply by BrightDV** [reply]
You can open the webview by enabling the Experimental Features from the settings, and then tap on Article Test Screen in the left drawer.

**Reply by ltguillaume** [reply]
I went to planetf1.com via the Article Test Screen and cannot reproduce the issue with this new version, so I think this can be closed 🙂 As soon as the new season starts, I'll do some additional tests during sessions.

Thanks again!
